lib/interface-types/src/serde/ser.rs

```diff
@@ -432,7 +441,9 @@ impl<'a> ser::SerializeStruct for &'a mut Serializer {
     }
 
     fn end(self) -> Result<Self::Ok, Self::Error> {
-        let record = InterfaceValue::Record(self.pop()?);
+        let record = InterfaceValue::Record(
+            Vec1::new(self.pop()?).map_err(|_| Self::Error::RecordNeedsAtLeastOneField)?,
```

Comment:
Could also use something like `TryFrom` here which returns the correct error automatically

Reply:
Yes, that's a good idea! I'll do that in another PR, so that this one isn't blocked.